dev/release/02-source.sh

```diff
@@ -202,7 +202,7 @@ The vote will be open for at least 72 hours.
 [10]: https://apache.jfrog.io/artifactory/arrow/python-rc/${version}-rc${rc}
 [11]: https://apache.jfrog.io/artifactory/arrow/ubuntu-rc/
 [12]: https://github.com/apache/arrow/blob/${release_hash}/CHANGELOG.md
-[13]: https://cwiki.apache.org/confluence/display/ARROW/How+to+Verify+Release+Candidates
+[13]: https://arrow.apache.org/docs/developers/release_verification.html
```

Comment:
We probably can change that. I will try to remember to update the vote thread email before sending it to point to the `dev` documentation for the 16.0.0 release.